src/libraries/System.Private.Xml/src/System/Xml/XmlCharType.cs

```diff
@@ -39,77 +39,22 @@ internal unsafe struct XmlCharType
         // bitmap for public ID characters - 1 bit per character 0x0 - 0x80; no character > 0x80 is a PUBLIC ID char
         private const string s_PublicIdBitmap = "\u2400\u0000\uffbb\uafff\uffff\u87ff\ufffe\u07ff";
 
-        // size of XmlCharType table
-        private const uint CharPropertiesSize = (uint)char.MaxValue + 1;
 
-        // static lock for XmlCharType class
-        private static object? s_Lock;
-
-        private static object StaticLock
-        {
-            get
-            {
-                if (s_Lock == null)
-                {
-                    object o = new object();
-                    Interlocked.CompareExchange<object?>(ref s_Lock, o, null);
-                }
-
-                return s_Lock;
-            }
-        }
-
-        private static volatile byte* s_CharProperties;
-        internal byte* charProperties;
-        private static void InitInstance()
-        {
-            lock (StaticLock)
-            {
-                if (s_CharProperties != null)
-                {
-                    return;
-                }
-
-                UnmanagedMemoryStream memStream = (UnmanagedMemoryStream)typeof(XmlWriter).Assembly.GetManifestResourceStream("XmlCharType.bin")!;
-                Debug.Assert(memStream.Length == CharPropertiesSize);
-
-                byte* chProps = memStream.PositionPointer;
-                Thread.MemoryBarrier();  // For weak memory models (IA64)
-                s_CharProperties = chProps;
-            }
-        }
-
-        private XmlCharType(byte* charProperties)
-        {
-            Debug.Assert(s_CharProperties != null);
-            this.charProperties = charProperties;
-        }
-
-        public static XmlCharType Instance
-        {
-            get
-            {
-                if (s_CharProperties == null)
-                {
-                    InitInstance();
-                }
-                return new XmlCharType(s_CharProperties);
-            }
-        }
+        public static XmlCharType Instance => default;
```

Comment:
This is unnecessary now. All methods on this type can be static methods now.

Reply:
They can but that's a much bigger change and all I was looking to do in this PR was replace the complicated resource loading code with the much simpler embedded span version. Refactoring this type and flowing those changes through the library can be done separately.

Reply:
You may need to do this refactoring to avoid regressing perf. Let's wait for the numbers...

Reply:
In the current version a byte* was being used to access the array as a single block. It was changed from previous versions where multiple arrays were being used. My assumption was that the dual layer caused two memory fetches that weren't serializable because the result of the first dictated the second. A single block means you get one fault of the correct page. My interpretation is that I've just replaced a byte* with a span and that the same lookup semantic of a single fault should remain in place.

Perf numbers would be good. Any guidance on getting them? is this a scenario that the performance repo covers? I wasn't looking to do a deep dive on this PR I just saw your comment that a replacement of the resource with a compiler backed version would be welcomed and thought it would be a fairly simply task to do.

Reply:
You may use any Xml sample for reading and writing. here is one example https://docs.microsoft.com/en-us/dotnet/api/system.xml.xmltextreader.read?view=netcore-3.1. and then use dotnet Benchmark library to measure the perf. let me know if you need help if you are not familiar with the perf library.